internal/grafanadi/handler/podsummary_handler.go

```diff
@@ -0,0 +1,211 @@
+package handler
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"sync"
+	"time"
+
+	"github.com/alipay/container-observability-service/internal/grafanadi/model"
+	"github.com/alipay/container-observability-service/pkg/dal/storage-client/data_access"
+	storagemodel "github.com/alipay/container-observability-service/pkg/dal/storage-client/model"
+
+	"github.com/alipay/container-observability-service/pkg/slo"
+	"github.com/alipay/container-observability-service/pkg/utils"
+	interutils "github.com/alipay/container-observability-service/internal/grafanadi/utils"
+	"github.com/alipay/container-observability-service/pkg/metrics"
+	"github.com/alipay/container-observability-service/internal/grafanadi/service"
+)
+
+type PodSummaryHandler struct {
+	request       *http.Request
+	writer        http.ResponseWriter
+	requestParams *PodSummaryParams
+	storage       data_access.StorageInterface
+}
+
+type PodSummaryParams struct {
+	PodUIDName string
+	PodUID     string
+}
+
+func (handler *PodSummaryHandler) RequestParams() interface{} {
+
+	return handler.requestParams
+}
+
+func (handler *PodSummaryHandler) ParseRequest() error {
+	params := PodSummaryParams{}
+	if handler.request.Method == http.MethodGet {
+		params.PodUIDName = handler.request.URL.Query().Get("searchkey")
+		params.PodUID = handler.request.URL.Query().Get("searchvalue")
+	}
+
+	handler.requestParams = &params
+	return nil
+}
+
+func (handler *PodSummaryHandler) ValidRequest() error {
```

Comment:
Isn’t parameter verification required here?

Reply:
Thank you very much for your comment. I am writing code here to imitate the format of other existing files in the internal/gfanadi/handler project. Perhaps the author of this code has their own considerations? ```golang func (handler *PodPhaseHandler) ValidRequest() error {
	return nil
}` [https://github.com/alipay/container-observability-service/blob/container-lifecycle-summary-cross-arch/internal/grafanadi/handler/podphase_handler.go](url)